File: src/main/java/com/wikia/tibia/repositories/StreetRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.StreetGateway;
import com.wikia.tibia.objects.Street;

public class StreetRepository extends WikiObjectRepository {

    public StreetRepository() {
        super(Street.class, new StreetGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/StreetRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Street

class StreetRepository : WikiObjectRepository<Street>(
        wikiObjectClass = Street::class.java,
        wikiObjectGateway = WikiObjectGateway(Contract.STREETS)
)